modules/components/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/DatasetApiV10.java

```diff
@@ -351,4 +344,46 @@ public Response getClusterPoints(@PathParam("datasetId") long datasetId,
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
         }
     }
+
+    /**
+     * Delete the dataset of a given dataset ID.
+     */
+    @DELETE
+    @Path("/{datasetId}")
+    @Produces("application/json")
+    public Response deleteDataset(@PathParam("datasetId") long datasetId) {
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        int tenantId = carbonContext.getTenantId();
+        String userName = carbonContext.getUsername();
+        try {
+            datasetProcessor.deleteDataset(tenantId, userName, datasetId);
+            return Response.ok().build();
+        } catch (MLDataProcessingException e) {
+            logger.error(String.format(
+                    "Error occurred while deleting dataset [id] %s of tenant [id] %s and [user] %s . Cause: %s",
+                    datasetId, tenantId, userName, e.getMessage()));
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
```

Comment:
It might be better to return a JSON message without directly returning error text when an error occurred. Then it would be easier for the client to process the response.